src/modifiers.ts

```diff
@@ -92,190 +93,200 @@ function hasSpecialChars(str: string): boolean {
                             ? new RegExp(`^${selector}$`, 'gi')
                             : new RegExp(selector, 'gi')
                         : undefined;
                 };
-                if (isWhereCommandDirective(directive)) {
-                    const groupRegex = getPatternToMatch(directive.where.group);
-                    const parameterRegex = getPatternToMatch(directive.where['parameter-name']);
-                    const commandRegex = getPatternToMatch(directive.where.command);
-                    const parameterReplacer =
-                        directive.set !== undefined ? directive.set['parameter-name'] : undefined;
-                    const paramDescriptionReplacer =
-                        directive.set !== undefined
-                            ? directive.set['parameter-description']
-                            : undefined;
-                    const commandReplacer =
-                        directive.set !== undefined ? directive.set.command : undefined;
-                    const commandDescriptionReplacer =
-                        directive.set !== undefined
-                            ? directive.set['command-description']
-                            : undefined;
-                    const groupReplacer =
-                        directive.set !== undefined ? directive.set.group : undefined;
-                    const groupDescriptionReplacer =
-                        directive.set !== undefined
-                            ? directive.set['group-description']
-                            : undefined;
-                    for (const operationGroup of values(this.codeModel.operationGroups)) {
-                        // operationGroup
-                        let groupChanged = false;
+                const groupRegex = getPatternToMatch(directive.where.group);
+                const parameterRegex = getPatternToMatch(directive.where['parameter-name']);
+                const commandRegex = getPatternToMatch(directive.where.command);
+                const parameterReplacer =
+                    directive.set !== undefined ? directive.set['parameter-name'] : undefined;
+                const paramDescriptionReplacer =
+                    directive.set !== undefined
+                        ? directive.set['parameter-description']
+                        : undefined;
+                const commandReplacer =
+                    directive.set !== undefined ? directive.set.command : undefined;
+                const commandDescriptionReplacer =
+                    directive.set !== undefined ? directive.set['command-description'] : undefined;
+                const groupReplacer = directive.set !== undefined ? directive.set.group : undefined;
+                const groupDescriptionReplacer =
+                    directive.set !== undefined ? directive.set['group-description'] : undefined;
+                let groupIdx = -1;
+                this.codeModel.operationGroups.forEach((operationGroup) => {
```

Comment:
can we split the detailed logic into different small functions?

Reply:
seperate it into three functions operationGroupModify operationModify and paramModify